### ISSUE TITLE
How to compute the epsilon achieved if the noise level dynamically changes for each training step? 

### ISSUE THREAD

**Issue by AshiakerWang** [woodyx218/Deep-Learning-with-GDP-Tensorflow#4]
For the current implementation of the function computing the epsilon achieved, the parameter noise level is fixed during the training process. I wonder how to compute the epsilon achieved after training if the noise level dynamically changes for each training step?

**Comment by woodyx218** [comment]
To compose with different levels of noises, you can use the f-DP, i.e. numerically compose (via Fourier transformation) the trade-off functions. However, GDP falls short for this case because central limit theorem requires iid samples.

**Comment by woodyx218** [comment]
On the other hand, if you have a step-wise weight decay scheme, e.g. lr=0.01 for epoch<100, lr=0.005 for epoch>100, then you may use GDP for each of the periods and numerically compose once at the intersection.

**Comment by AshiakerWang** [comment]
I sincerely appreciate your explanation, it helps a lot!
